src/main.rs

```diff
@@ -36,11 +51,29 @@ struct TypeSizeCmd {
 
 impl TypeSizeCmd {
     fn new(old_args: impl Iterator<Item = String>) -> Self {
-        let cargo_subcommand = "check";
+        let mut cargo_subcommand = "check";
         let args = vec![];
         let mut typesize_args: Vec<String> = vec![];
 
-        typesize_args.extend(old_args);
+        for arg in old_args {
+            match arg.as_str() {
+                "--build" => {
```

Comment:
I think I prefer without the `--` in `--<subcmd>` i.e. `cargo typesize build`,  `cargo typesize test`,  `cargo typesize check`

It looks closer to cargo when you start adding options to the subcommands e.g.

`cargo typesize build --example <example_name>` vs
`cargo typesize --build --example <example_name>`